src/data/finsupp/basic.lean

```diff
@@ -1483,6 +1483,50 @@ end
 
 end comap_domain
 
+section option
+
+/-- Restrict a finitely supported function on `option α` to a finitely support function on `α`. -/
+def some [has_zero M] (f : option α →₀ M) : α →₀ M :=
+f.comap_domain option.some (by tidy)
+
+@[simp] lemma some_apply [has_zero M] (f : option α →₀ M) (a : α) :
+  f.some a = f (option.some a) := rfl
+
+@[simp] lemma zero_sum [has_zero M] : (0 : option α →₀ M).some = 0 :=
+by { ext, simp, }
+
+@[simp] lemma add_sum [add_comm_monoid M] (f g : option α →₀ M) : (f + g).some = f.some + g.some :=
+by { ext, simp, }
+
+@[simp] lemma single_none_some [has_zero M] (m : M) : (single none m : option α →₀ M).some = 0 :=
+by { ext, simp, }
+
+@[simp] lemma single_some_some [has_zero M] (a : α) (m : M) :
+  (single (option.some a) m : option α →₀ M).some = single a m :=
+by { ext b, simp [single_apply], }
+
+lemma sum_option [add_comm_monoid M] [add_comm_monoid N]
+  (f : option α →₀ M) (b : option α → M →+ N) :
```

Comment:
I think it makes sense to state this without a bundled hom, as we do for `sum_add_index` - making this probably `sum_option_index`.

Also this should be `to_additive` I think.

Reply:
I like the name `sum_option_index`, but I don't think that unbundling is worth it here. It would be quite long-winded, and this is the version I need down the track.

Reply:
Similarly, the multiplicative version is cumbersome to state, and I don't need it, so I'm disinclined to add it here.

Reply:
The multiplicative version is only cumbersome to state because you bundled here. We have both `sum_add_index` and `prod_add_index`, so we should have both versions for `option` too.